includes/AJAX.php

```diff
@@ -33,6 +35,56 @@ public function __construct() {
 
 		// maybe output a modal prompt when setting excluded terms
 		add_action( 'wp_ajax_facebook_for_woocommerce_set_excluded_terms_prompt', [ $this, 'handle_set_excluded_terms_prompt' ] );
+
+		// sync all products via AJAX
+		add_action( 'wp_ajax_wc_facebook_sync_products', [ $this, 'sync_products' ] );
+
+		// get the current sync status
+		add_action( 'wp_ajax_wc_facebook_get_sync_status', [ $this, 'get_sync_status' ] );
+	}
+
+
+	/**
+	 * Syncs all products via AJAX.
+	 *
+	 * @internal
+	 *
+	 * @since 2.0.0-dev.1
+	 */
+	public function sync_products() {
+
+		check_admin_referer( Product_Sync::ACTION_SYNC_PRODUCTS, 'nonce' );
+
+		facebook_for_woocommerce()->get_products_sync_handler()->create_or_update_all_products();
+
+		wp_send_json_success();
+	}
+
+
+	/**
+	 * Gets the current sync status.
+	 *
+	 * @internal
+	 *
+	 * @since 2.0.0-dev.1
+	 */
+	public function get_sync_status() {
+
+		check_admin_referer( Product_Sync::ACTION_GET_SYNC_STATUS, 'nonce' );
+
+		$remaining_products = 0;
+		$jobs               = facebook_for_woocommerce()->get_products_sync_background_handler()->get_jobs( [
+			'status' => 'processing',
+		] );
+
+		if ( ! empty( $jobs ) ) {
+
+			foreach ( $jobs as $job ) {
+				$remaining_products = $job->total - $job->progress;
```

Comment:
Should we add to the `$remaining_products` or are we intentionally only getting the latest? If that is the case, it would be better to just get the latest instead of iterating all processing jobs.

Reply:
good call - in fact there is also a case where the total & progress may not be set yet. I tweaked things here to be a little more solid: dfc7f918063ea62271ac20b711768aa01d66ba4b